.github/workflows/release.yaml

```diff
@@ -1,16 +1,13 @@
-# Release workflow: Pushing to a release branch triggers this job, which
-# builds a Docker image, tags it to whatever is specified in the .version
-# file, and pushes it to the GitHub Container Registry (GHCR).
-# For example, if specifying 0.2.3 in the .version file then pushing that
-# change to a release branch, an image with ghcr.io/lustre-csi-driver:0.2.3
-# would be built and made available via Packages.
+# Release workflow: Pushing a tag triggers this workflow,
+# which uses the name of the tag as the container tag/version
+# that goes into GHCR.
 name: Release
 
+# Build and publish Docker image upon pushing a tag
 on:
   push:
-    # Build and publish package for release branches
-    branches:
-      - "release-**"
+    tag:
+      - "*"
```

Comment:
You might want to make this "v*", since by default git-version-gen expects that to be the version tag prefix, and is going to ignore any tags that don't start with that. The prefix could be something else, but we would need to add the "--prefix" option to the command line of git-version-gen to select something else.